prime-router/src/test/kotlin/tokens/AuthenticationStrategyTests.kt

```diff
@@ -125,49 +127,43 @@ class AuthenticationStrategyTests {
+    fun `test validate claims`() {
         val req = MockHttpRequestMessage("test")
-        req.httpHeaders["Authorization"] = "Bearer tok1"
-        var tok = AuthenticationStrategy.getAccessToken(req)
-        assertThat(tok).isEqualTo("tok1")
-
-        req.httpHeaders.clear()
-        req.httpHeaders["authorization"] = "Bearer tok2"
-        tok = AuthenticationStrategy.getAccessToken(req)
-        assertThat(tok).isEqualTo("tok2")
-
-        req.httpHeaders.clear()
-        req.httpHeaders["AUTHORIZATION"] = "Bearer tok3"
-        tok = AuthenticationStrategy.getAccessToken(req)
-        assertThat(tok).isEqualTo("tok3")
-
-        req.httpHeaders.clear()
-        req.httpHeaders["authorization"] = "tok4"
-        tok = AuthenticationStrategy.getAccessToken(req)
-        assertThat(tok).isEqualTo("tok4")
-
-        req.httpHeaders.clear()
-        req.httpHeaders["foobar"] = "Bearer tok5"
-        tok = AuthenticationStrategy.getAccessToken(req)
-        assertThat(tok).isNull()
-
-        req.httpHeaders.clear()
-        tok = AuthenticationStrategy.getAccessToken(req)
-        assertThat(tok).isNull()
-
-        req.httpHeaders.clear()
-        req.httpHeaders["foobar"] = "Bearer tok5"
-        tok = AuthenticationStrategy.getAccessToken(req)
-        assertThat(tok).isNull()
-
-        req.httpHeaders.clear()
-        req.httpHeaders["foobar"] = "bearer tok5"
-        tok = AuthenticationStrategy.getAccessToken(req)
-        assertThat(tok).isNull()
-
-        req.httpHeaders.clear()
-        req.httpHeaders["foobar"] = "BEARER tok5"
-        tok = AuthenticationStrategy.getAccessToken(req)
-        assertThat(tok).isNull()
+        // Missing authentication-type header in req!  Means ==> use token auth.
+        val jwtClaims = mockkClass(Claims::class)
+        every { jwtClaims[any()] } returns "simple_report.default.report"
+        mockkConstructor(TokenAuthentication::class)
+        every { anyConstructed<TokenAuthentication>().authenticate(any(), any()) } returns jwtClaims
+
+        val claims = AuthenticationStrategy.authenticate(req)
+        val mismatchedSender = CovidSender(
+            "Test Sender",
+            "test",
+            Sender.Format.HL7,
+            schemaName =
+            "one",
+            allowDuplicates = true
+        )
+        val matchingSender = CovidSender(
+            "Test Sender",
+            "simple_report",
+            Sender.Format.HL7,
+            schemaName =
+            "one",
+            allowDuplicates = true
+        )
+
+        if (claims != null) {
+            var result = AuthenticationStrategy.validateClaim(claims, mismatchedSender, req)
+            assertThat(result).isFalse()
+
+            result = AuthenticationStrategy.validateClaim(claims, matchingSender, req)
+            assertThat(result).isTrue()
+
+            // TODO: mock claims in a way so as to NOT change isPrimeAdmin from 'val' to 'var' in AuthenticatedClaims.kt
+            claims.isPrimeAdmin = true
```

Comment:
I couldn't immediately figure out how to custom mock a Claim with the "isPrimeAdmin" set to true, so I used this egregious cheat for the short term (in the interests of getting this PR up). Will follow up tomorrow and fix before merging. Perhaps you, dear reader, could point me in the right direction?

Reply:
fixed it